control/plugin/client/grpc.go

```diff
@@ -123,7 +125,8 @@ func SecurityTLSOff() GRPCSecurity {
 
 // NewCollectorGrpcClient returns a collector gRPC Client.
 func NewCollectorGrpcClient(address string, timeout time.Duration, security GRPCSecurity) (PluginCollectorClient, error) {
-	p, err := newPluginGrpcClient(address, timeout, security, plugin.CollectorPluginType)
+	ctx := context.Background()
+	p, err := newPluginGrpcClient(ctx, address, timeout, security, plugin.CollectorPluginType)
```

Comment:
@rashmigottipati, does it have sense to move creating context (as in line https://github.com/intelsdi-x/snap/pull/1681/files#diff-706e573cd7a189d87a3edea5ee3ed0f4R128) from `NewCollectorGrpcClient`  to `newPluginGrpcClient()` or even lower to `newGrpcClient()`? Is this possible to avoid passing ctx cross all layers?